iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -575,33 +592,71 @@ private void commitWrite(Table table, String branchName, Long snapshotId, long s
   }
 
   /**
-   * Creates and commits an Iceberg insert overwrite change with the provided data files.
-   * For unpartitioned tables the table content is replaced with the new data files. If not data files are provided
-   * then the unpartitioned table is truncated.
-   * For partitioned tables the relevant partitions are replaced with the new data files. If no data files are provided
-   * then the unpartitioned table remains unchanged.
-   * @param table The table we are changing
-   * @param startTime The start time of the commit - used only for logging
-   * @param results The object containing the new files
-   * @param rewritePolicy The rewrite policy to use for the insert overwrite commit
+   * Creates and commits an Iceberg compaction change with the provided data files.
+   * Either full table or a selected partition contents is replaced with compacted files.
+   *
+   * @param table             The table we are changing
+   * @param startTime         The start time of the commit - used only for logging
+   * @param results           The object containing the new files
+   * @param rewritePolicy     The rewrite policy to use for the insert overwrite commit
+   * @param partitionSpecId   The table spec_id for partition compaction operation
+   * @param partitionPath     The path of the compacted partition
    */
-  private void commitOverwrite(Table table, String branchName, long startTime, FilesForCommit results,
-      RewritePolicy rewritePolicy) {
+  private void commitCompaction(Table table, long startTime, FilesForCommit results,
+      RewritePolicy rewritePolicy, Integer partitionSpecId, String partitionPath) {
     Preconditions.checkArgument(results.deleteFiles().isEmpty(), "Can not handle deletes with overwrite");
     if (!results.dataFiles().isEmpty()) {
```

Comment:
no need for nested if, return right away if `results.dataFiles().isEmpty()`

Reply:
Done